### DISCUSSION TITLE
choice of video resolution desired

### DISCUSSION THREAD

**Discussion by lukerg** [gumcast/gumcast-client#28] [Ideas]
I've not yet dug into the code to see how at all this might be possible, but will try to do that later.
In brief, I would like a way to influence which video file is returned in the feed that gumcast returns. For quota reasons, I would prefer the lower resolution videos to be returned, whenever there is such a choice. I can see from the gumroad web player there is a choice (when the creator has made two or more resolutions).
Right now, it appears to be the case that gumcast is only returning the 1080p resolution videos for the given feed I am concerned with. For what it is worth, the 1080p result is the second one in the m3u8 file that the gumroad web api returns, suggesting that gumcast is picking the 1080p one deliberately.
I made this discussion as a tracking point for a possible feature, assuming this does not already exist - I couldn't see anything listed in the FAQ about how to do this.

**Comment by bcomnes** [comment]
Gumcast returns a feed of all files in a gumroad product ordered by date. Generally, products on gumroad are audio or video files, served directly off the gumroad API file storage backend (same thing the Gumroad apps use). 

I have not seen any products use m3u8 HLS chunked/vbr video streams. Are they supporting that now? If this is the case, GumCast could support redirecting to these streams, but YMMV when trying to play them in different podcast apps. From testing the GumCast concept on a few other sources that do have HLS video streams, I know apple podcasts will play them, but won't go full screen on them for some reason. They work really well on Apple TV though. Apple's podcast specs are fairly clear that you need to be serving URLs that resolve to full video/audio files from a host that supports range requests for streaming seeking. HLS support in podcast feeds is basically a nice bug it seems. To add proper HLS support to GumCast allowing people to download a copy of the stream would require charging people money for compute/storage costs of converting, storing and serving de-chunked video files that podcast apps expect. I call out Apple here, because they are basically the lowest common denominator when it comes to what can be done in podcast feeds.

I have seen products/creators upload multiple format sizes of the same video to the product though. Maybe this is what you have seen? 

Either way, other than some basic MIME type info, there isn't a field provided for Gumcast to select or filter for these different versions/sizes iirc, and the file naming conventions varies from creator to creator, so there isn't really a good common pattern that exists I could follow. 

The best idea I could come up with is some kind of query string that could be added that a user can opt in that performs some kind of filtering on the feed. Any idea what that would look for?

**Reply by bcomnes** [reply]
I know gumroad added folders which gumcast has not added any kind of support for. Would a feed with just the contents of a folder help you?

**Reply by bcomnes** [reply]
I just looked at the web player and it looks like they do have some kind of HLS stream support for that, so at some point they are chunking the video at different sizes. Must be new. I'm guessing they don't make available the re-sized versions as whole files though, which is necessary for a proper podcast feed.

**Reply by lukerg** [reply]
it seems i got my wires a bit crossed here. I see now that what the web player offers is different to what the API spits out. The end device I am using here is a kodi HTPC rather than a mobile device, which has thus far accepted the single file mp4/webm files from the gumcast feed I am interested in.

I am not sure the folders make any difference for what I am after here. The feed I am looking at does use folders but it is not at all important to me - the creator just uploads the videos and leaves it at that.

May I ask how/where you figured out the mobile API?  I will try to dig around in the data feed i get to see if there is something that can be matched upon via a query string as you suggested.

**Reply by bcomnes** [reply]
The content api appears reserved for the gumroad mobile app. Everything here was built by intercepting network traffic and looking at the mobile app artifacts to figure out how to build the feed, so everything I can share is in here: https://github.com/gumcast/gumcast-api
